libs/dataviews-client/src/resolve-dataviews-generators.ts

```diff
@@ -442,12 +442,22 @@ export function getDataviewsGeneratorConfigs(
       return sublayer
     })
 
+    const maxZoomLevels = dataviews
+      ?.filter(({ config }) => config && config?.maxZoom)
+      .flatMap(({ config }) => config?.maxZoom)
     const mergedActivityDataview = {
       id: params.mergedActivityGeneratorId || MERGED_ACTIVITY_ANIMATED_HEATMAP_GENERATOR_ID,
       config: {
         type: GeneratorType.HeatmapAnimated,
         sublayers: activitySublayers,
         mode: heatmapAnimatedMode,
+        // if any of the activity dataviews has a max zoom level defined
+        // apply the minimum max zoom level (the most restrictive approach)
+        ...((maxZoomLevels &&
+          maxZoomLevels.length > 0 && {
+            maxZoom: Math.min(...maxZoomLevels),
+          }) ||
```

Comment:
do we need the or condition ?
I think if none of `maxZoomLevels && maxZoomLevels.length > 0` it won't add anything into the config anyway